tests/test_download_utilities.py

```
@@ -0,0 +1,437 @@
+    def test_atomic_write_success(self, tmp_path):
+        """Test successful atomic write."""
+        test_file = tmp_path / "test.txt"
+        content = b"test content"
+
+        result = downloader._atomic_write(str(test_file), content)
+
+        assert result is True
+        assert test_file.exists()
+        assert test_file.read_bytes() == content
+
+    def test_atomic_write_failure(self, tmp_path):
```

Ruff (0.14.2)

Unused method argument: `tmp_path`

(ARG002)